[WIP] New action `validate_play_store_json_key`

janpio opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Using _supply_ requires a [setup step](https://docs.fastlane.tools/actions/supply/#setup) which includes a json file for authentication with Google Play Store. This is error prone and difficult to test without 

### Description
Implement `validate_play_store_json_key` that uses `Supply::client` to check if `json_key` can be used to establish a working connection to the Google Play API.

### Tasks

- [ ] Tests
- [ ] Check if it actually works
- [ ] Add to supply documentation

janpio committed:
Move over actions from plugin with minimal changes

janpio committed:
create empty spec files for actions

janpio committed:
remove unneeded/broken require

janpio committed:
update author

janpio committed:
return URI in get_managedplay_store_publishing_rights

janpio committed:
make clear the details will be filled

janpio committed:
add action category

janpio committed:
make clear that URL will change

janpio committed:
example_code for get_managed_play_store_publishing_rights

janpio committed:
include documentation from README into details of get_managed_play_store_publishing_rights

janpio committed:
add description and example_code for create_app_on_managed_play_store

janpio committed:
better example

janpio committed:
fix code example

janpio committed:
remove root_url and timeout for now

janpio committed:
remove TODOs that were added to PR

janpio committed:
replace callback_uri with probable future URL

janpio committed:
rubocop

janpio committed:
rubocop 2

janpio committed:
fix order in verify_block for Supply's json_key:
check if file exists first, then if it is JSON

janpio committed:
Extract reusable parts of Client into GoogleServiceClient (self.make_from_config, initialize, call_google_api)

janpio committed:
Modify create_app_on_managed_play_store to implement and use Supply::GoogleServiceClient

janpio committed:
remove leftover code and TODOs
rubocop

janpio committed:
skip fastlane/spec/unused_options_spec.rb for new action

janpio committed:
remove TODO

janpio committed:
Implement validate_play_store_json_key that uses Supply::client to check if connection works

fastlane-bot-helper commented:
<!--
  0 Errors
  1 Warning: Pull Request is Work in Progre...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Pull Request is Work in Progress</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

janpio committed:
add validate_play_store_json_key to unused_options list

janpio committed:
fix rubocop

janpio committed:
Rename Supply::GoogleServiceClient to Supply::AbstractGoogleServiceClient

janpio committed:
replce @scope/@service with SCOPE/SERVICE and use self.class::SCOPE/SERVICE to read it
https://stackoverflow.com/questions/19382626/how-do-i-override-a-variable-in-a-ruby-subclass-without-affecting-the-superclass/19382692#19382692

janpio committed:
workaround for `warning: already initialized constant Supply::UI`

janpio committed:
get rid of more code by moving client/android_publisher to AbstractGoogleServiceClient as client
(and also get rid of empty method implementations in PlaycustomappClient)
